src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp

```diff
@@ -495,19 +494,27 @@ namespace OpenMS
     OPENMS_LOG_INFO << seeds_added << " seeds without RT and m/z overlap with existing IDs added" << endl;
   }
 
-
   pair<String, Int> FeatureFinderIdentificationAlgorithm::extractTargetID_(
     const Feature& feature, bool extract_charge)
   {
     String target_id = feature.getMetaValue("PeptideRef");
-    Size pos_slash = target_id.rfind('/');
+
```

Comment:
I think there was a bug. @oliveralka knows more.

Reply:
PEP:XXXXX/2  did lead to an error since no was hash could be found in the string.
That is why I added two options (if/else). 
This could probably be handled much cleaner with a more complicated parsing as Timo mentioned once.

Reply:
> PEP:XXXXX/2 did lead to an error since no was hash could be found in the string.

That can't be true. If there's no hash, the `find` call returns `npos` and the `substr` call reads to the end of the string. So you get the correct result.

Reply:
Seems it was the other way round, sorry I will double-check:

```
Error: Unexpected internal error (Prefix of string '3#1' successfully converted to an integer value. Additional characters found at position 2)
<While loading '/Users/alka/Documents/work/software/openms/OpenMS/share/OpenMS/examples/FRACTIONS/BSA1_F2.mzML': Obsolete CV term 'MS:1000763 - PeakPicker' used in tag 'software'.> occurred 2 times
```

Reply:
Okay, there was indeed a bug, but for reference strings *with* a hash, e.g. "PEP:XXXXX/2#1". The reason was the second argument in the `substr` call: That should be a character count, *not* the end position of the substring. I've fixed it.

Reply:
thank you!